sklearn/metrics/_classification.py

```diff
@@ -909,6 +938,15 @@ class labels [2]_.
     >>> cohen_kappa_score(y1, y2)
     0.6875
     """
+
+    def _check_zero_division(denominator, replace_undefined_by, msg):
+        if np.isclose(denominator, 0):
+            if replace_undefined_by == "deprecated":
+                replace_undefined_by = np.nan
+                warnings.warn(mgs_changing_default, FutureWarning)
+            warnings.warn(msg, UndefinedMetricWarning, stacklevel=2)
```

Comment:
here you're modifying `replace_undefined_by` locally (in the scope of `_check_zero_division`), so it's not propagated outside of this function and the metric returns `"deprecated"`. You can return it to be able to use the new value outside of the function. Please also add a test.

Reply:
Actually I thought that the consensus was to set `np.nan` as default so there's probably no need for a FutureWarning at all.